recipes/coost/all/conanfile.py

```diff
@@ -0,0 +1,105 @@
+        "fPIC": [True, False],
+        "with_libcurl": [True, False],
+        "with_openssl": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_libcurl": False,
+        "with_openssl": False,
+    }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+
+    def requirements(self):
+        if self.options.with_libcurl:
+            self.requires("libcurl/7.80.0")
+        if self.options.with_libcurl or self.options.with_openssl:
+            self.requires("openssl/1.1.1q")
+
+    def validate(self):
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, 11)
+        if self.info.options.with_libcurl:
+            if not self.info.options.with_openssl:
+                raise ConanInvalidConfiguration(f"{self.ref} requires with_openssl=True when using with_libcurl=True")
+            if self.dependencies["libcurl"].options.with_ssl != "openssl":
+                raise ConanInvalidConfiguration(f"{self.ref} requires libcurl:with_ssl='openssl' to be enabled")
+            if not self.dependencies["libcurl"].options.with_zlib:
+                raise ConanInvalidConfiguration(f"{self.ref} requires libcurl:with_zlib=True to be enabled")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        if is_msvc(self):
+            tc.variables["STATIC_VS_CRT"] = is_msvc_static_runtime(self)
+        tc.variables["WITH_LIBCURL"] = self.options.with_libcurl
+        tc.variables["WITH_OPENSSL"] = self.options.with_openssl
+        tc.generate()
+        cd = CMakeDeps(self)
+        cd.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, "LICENSE.md", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "coost")
+        self.cpp_info.set_property("cmake_target_name", "coost::co")
+        # TODO: back to global scope in conan v2 once legacy generators removed
+        self.cpp_info.components["co"].libs = ["co"]
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            self.cpp_info.components["co"].system_libs = ["pthread", "dl"]
```

Comment:
```suggestion
            self.cpp_info.components["co"].system_libs = ["pthread", "dl"]
        elif self.settings.os == "Windows":
            self.cpp_info.components["co"].system_libs = ["ws2_32"]
```
Windows needs ws2_32

Reply:
Thanks!